examples/adjoints/adjoints_ex5/heatsystem.C

```diff
@@ -62,8 +62,10 @@ void HeatSystem::init_data ()
 
   ZeroFunction<Number> zero;
 
+  // Most DirichletBoundary users should use this style of
+  // constructor, which takes a "locally indexed" functor
   this->get_dof_map().add_dirichlet_boundary
-    (DirichletBoundary (all_bdys, T_only, &zero));
+    (DirichletBoundary (zero, all_bdys, T_only));
```

Comment:
So, from this diff it seems that the new constructors take the same arguments, just in a different order and as references rather than pointers?  This is probably going to forever confuse me/have me looking at the docs to see which order is which... would it not be possible to have the constructor take an `enum LOCALLY_INDEXED` or something so that it's clear from reading the code what indexing you are using?